### DISCUSSION TITLE
Question about QHeaderView

### DISCUSSION THREAD

**Discussion by RealLucioCarreras** [tsujan/Kvantum#626] [Q&A]
Hello,

first of all, thank for your great work! I am a Gnome user and really appreciated the things you did to improve UE.

I have a question regarding the `QHeaderView`. When trying to apply a CSS stylesheet to the header view I am only able to set the background color but not the foreground color. It seems that the foreground color is the same as in the default theme and the css value is ignored.

e.g.
```
QHeaderView::section
{
   background: red;
   color: white;
}
```
The white color is still ignored but the background is painted red. Do you have any hints/workarounds for me?

Thanks in advance

**Comment by tsujan** [comment]
I'm afraid your question is about Qt stylesheets, not Kvantum. Kvantum isn't related to CSS.

Usually, if a stylesheet is complete enough, it'll override the widget style (Kvantum, in this case), not conversely. I guess the above CSS block isn't enough for a stylesheet intrusion; you probably need more.

There is an old discrepancy between style sheets and widget styles is in Qt. For example, see https://bugreports.qt.io/browse/QTBUG-68671 (which isn't fixed in Qt6).

**Reply by RealLucioCarreras** [reply]
Thanks you for the fast reply

I also discovered some issues in Qt's stylesheet mechanism which I had to do some workarounds. The stylesheet is in fact way more complex than the one I posted :wink: 

I was just wondering because the above mentioned issue only appears when using Kvantum. With Gtk2 or Plasma theme this does not happen and I can style the header view as I want to.

But there's a workaround I can apply. I was just curious if there's a way to solve this more elegantly

**Reply by tsujan** [reply]
> I was just wondering because the above mentioned issue only appears when using Kvantum

That's possible. Each widget style may have a different way of drawing widgets. But how or when a Qt stylesheet overrides the widget style is outside Kvantum's control.

Actually, the real problem is the opposite. A developer may use stylesheets in his code but forget how far they can affect the widget style. Then, you may see their effects where they shouldn't be present. For example, the result can be a wrong layout, where some widgets don't have enough space. That's a bad way of coding but Qt allows it.

I hoped that, by fixing https://bugreports.qt.io/browse/QTBUG-68671, Qt would be more consistent in this regard but that didn't happen.
